phpstan.neon.dist

```diff
@@ -120,8 +135,12 @@ parameters:
             paths:
                 - lib/Doctrine/DBAL/Driver/OCI8/OCI8Statement.php
 
-        # https://github.com/phpstan/phpstan-src/pull/700
         -
-            message: '~^Parameter #2 \$count of function array_fill expects int<0, max>, int given\.$~'
+            message: '~Method Doctrine\\DBAL\\Driver\\Mysqli\\MysqliStatement::rowCount\(\) should return int but returns int\|string\.~'
```

Comment:
@morozov This error seems to be raised because of the stub you contributed in /phpstan/phpstan-src#700.

The `mysqli_stmt::$affected_rows` property is documented as `int|string` which is also what the documentation on php.net says. The documentation does not say in which cases we can expect a string here, but my guess is that this happens if the number of affected rows is larger than `PHP_INT_MAX`.

However, this also applies to `mysqli_stmt::$num_rows` which your stub documents as `0|positive-int`.

Reply:
So the stub probably should be `0|positive-int|string`, as well as the return type in the DBAL driver.

Reply:
But it looks like we also need to change the `@return` annotation on `(Statement|Result)::rowCount()` from `int` to `int|string` because it will return a string if the value is greater than `PHP_INT_MAX`.

Reply:
This is a bit of a problem because on 3.1.x, we already have an `int` return type declaration.

Reply:
I see. Then I think it's acceptable. Unlike the discussion in https://github.com/doctrine/orm/issues/8963#issuecomment-909564835, updating or deleting a number of rows larger than `PHP_INT_MAX` is close to impossible.